packages/storage-driver-cloudinary/src/index.ts

```diff
@@ -0,0 +1,387 @@
+		return `Basic ${base64}`;
+	}
+
+	async read(filepath: string, range?: Range) {
+		const resourceType = this.getResourceType(filepath);
+		const fullPath = this.fullPath(filepath);
+		const signature = this.getParameterSignature(fullPath);
+		const url = `https://res.cloudinary.com/${this.cloudName}/${resourceType}/upload/${signature}/${fullPath}`;
+
+		const requestInit: RequestInit = { method: 'GET' };
+
+		if (range) {
+			requestInit.headers = {
+				Range: `bytes=${range.start ?? ''}-${range.end ?? ''}`,
+			};
+		}
+
+		const response = await fetch(url, requestInit);
+
+		if (response.status >= 400 || !response.body) {
+			throw new Error(`No stream returned for file "${filepath}"`);
+		}
+
+		return Readable.fromWeb(response.body);
+	}
+
+	async stat(filepath: string) {
+		const fullPath = this.fullPath(filepath);
+		const resourceType = this.getResourceType(fullPath);
+		const publicId = this.getPublicId(fullPath);
+		const url = `https://api.cloudinary.com/v1_1/${this.cloudName}/resources/${resourceType}/upload/${publicId}`;
+		const response = await fetch(url, {
+			method: 'GET',
+			headers: {
+				Authorization: this.getBasicAuth(),
+			},
+		});
+
+		if (response.status >= 400) {
+			throw new Error(`No stat returned for file "${filepath}"`);
+		}
+
+		const { bytes, created_at } = (await response.json()) as { bytes: number; created_at: string };
+		return { size: bytes, modified: new Date(created_at) };
+	}
+
+	async exists(filepath: string) {
+		try {
+			await this.stat(filepath);
+			return true;
+		} catch {
+			return false;
+		}
+	}
+
+	async move(src: string, dest: string) {
+		const fullSrc = this.fullPath(src);
+		const fullDest = this.fullPath(dest);
+		const resourceType = this.getResourceType(fullSrc);
+
+		const url = `https://api.cloudinary.com/v1_1/${this.cloudName}/${resourceType}/rename`;
+
+		const parameters = {
+			from_public_id: this.getPublicId(fullSrc),
+			to_public_id: this.getPublicId(fullDest),
+			api_key: this.apiKey,
+			timestamp: this.getTimestamp(),
+		};
+
+		const signature = this.getFullSignature(parameters);
+
+		const body = this.toFormUrlEncoded({
+			...parameters,
+			signature,
+		});
+
+		const response = await fetch(url, {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+			},
+			body,
+		});
+
+		if (response.status >= 400) {
+			const responseData = (await response.json()) as { error?: { message?: string } };
+			throw new Error(`Can't move file "${src}": ${responseData?.error?.message ?? 'Unknown'}`);
+		}
+	}
+
+	async copy(src: string, dest: string) {
+		const stream = await this.read(src);
+		await this.write(dest, stream);
+	}
+
+	async write(filepath: string, content: Readable) {
+		const fullPath = this.fullPath(filepath);
+		const resourceType = this.getResourceType(fullPath);
+
+		const timestamp = this.getTimestamp();
+
+		const uploadParameters = {
+			timestamp: timestamp,
```

Comment:
```suggestion
		const uploadParameters = {
			timestamp: this.getTimestamp(),
```

https://github.com/directus/directus/blob/05e4917f2161b9a3867831b9f479048847e32ccc/packages/storage-driver-cloudinary/src/index.ts#L175

Super tiny suggestion to make it more similar to how timestamp parameter is directly set in the `move()` method.

Reply:
You have a scary good eye for detail 😘 That's a leftover from an earlier state where the uploadchunk method was still within the same function context and it would reuse the function scope variable 🙂